src/main/java/com/exasol/projectkeeper/AbstractProjectFilesWalker.java

```diff
@@ -0,0 +1,95 @@
+package com.exasol.projectkeeper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.Resource;
+import io.github.classgraph.ScanResult;
+
+/**
+ * Abstract basis classes that validate the project files.
+ * <p>
+ * This class walks the project templates, and calls a {@link ProjectFileTemplateVisitor} for each template file.
+ * </p>
+ */
+public class AbstractProjectFilesWalker {
+    private static final Pattern TEMPLATE_NAME_PATTERN = Pattern
+            .compile(".*\\/templates\\/([^\\/]+)\\/([^\\/]+)\\/(.*)");
+
+    /**
+     * Run the template walking.
+     * 
+     * @param visitor          visitor to be called for each template
+     * @param projectDirectory project's root
+     * @param enabledModules   list of enabled modules
+     */
+    protected void run(final ProjectFileTemplateVisitor visitor, final File projectDirectory,
+            final List<String> enabledModules) {
+        try (final ScanResult scanResult = new ClassGraph().acceptPaths("templates/").scan()) {
+            scanResult.getAllResources().forEach(resource -> {
+                walkFile(visitor, projectDirectory, enabledModules, resource);
+            });
+        }
+    }
+
+    private void walkFile(final ProjectFileTemplateVisitor visitor, final File projectDirectory,
+            final List<String> enabledModules, final Resource resource) {
+        final String resourceName = resource.getURI().toString();
+        final Matcher matcher = TEMPLATE_NAME_PATTERN.matcher(resourceName);
+        if (!matcher.matches()) {
+            throw new IllegalStateException("Template name had invalid format.");
+        }
+        final String module = matcher.group(1);
+        final TemplateType templateType = templateTypeFromString(matcher.group(2));
+        final String fileName = matcher.group(3);
+        if (enabledModules.contains(module)) {
+            invokeVisitor(visitor, projectDirectory, resource, templateType, fileName);
+        }
+    }
+
+    private void invokeVisitor(final ProjectFileTemplateVisitor visitor, final File projectDirectory,
+            final Resource resource, final TemplateType templateType, final String fileName) {
+        try (final InputStream templateStream = resource.open()) {
+            final File projectFile = projectDirectory.toPath().resolve(fileName).toFile();
+            visitor.visit(fileName, projectFile, templateStream, templateType);
+        } catch (final IOException exception) {
+            throw new IllegalStateException("Failed to open template file.", exception);
+        }
+    }
+
+    private TemplateType templateTypeFromString(final String templateTypeString) {
```

Comment:
I think it's better to move this logic inside the enum.

Reply:
:+1: